google/cloud/aiplatform/metadata/artifact.py

```diff
@@ -41,24 +90,108 @@ def _create_resource(
         parent: str,
         resource_id: str,
         schema_title: str,
+        uri: Optional[str] = None,
         display_name: Optional[str] = None,
         schema_version: Optional[str] = None,
         description: Optional[str] = None,
         metadata: Optional[Dict] = None,
-    ) -> proto.Message:
+    ) -> gca_artifact.Artifact:
         gapic_artifact = gca_artifact.Artifact(
+            uri=uri,
             schema_title=schema_title,
             schema_version=schema_version,
             display_name=display_name,
             description=description,
             metadata=metadata if metadata else {},
+            state=gca_artifact.Artifact.State.LIVE,
```

Comment:
No specific handles. `LIVE` was selected because it is consistent with the default behavior of Pipelines. Open to changing the default state.

Reply:
I see, as this is for artifact tracking, I think the user could define their own use cases for the artifacts. So maybe instead of preset to `LIVE` or default, shall we add a parameter to allow user to specific the state? And we can default to `STATE_UNSPECIFIED` if state is not set?

Reply:
Updated.. Exposed the argument and defaulted to `STATE_UNSPECIFIED`.